mypy/plugins/attrs.py

```diff
@@ -1022,17 +1022,30 @@ def add_method(
         )
 
 
-def _get_attrs_init_type(typ: Instance) -> CallableType | None:
+def _get_attrs_init_type(typ: Instance) -> CallableType | None | AnyType:
     """
     If `typ` refers to an attrs class, get the type of its initializer method.
     """
     magic_attr = typ.type.get(MAGIC_ATTR_NAME)
     if magic_attr is None or not magic_attr.plugin_generated:
         return None
     init_method = typ.type.get_method("__init__") or typ.type.get_method(ATTRS_INIT_NAME)
-    if not isinstance(init_method, FuncDef) or not isinstance(init_method.type, CallableType):
+    if init_method is None:
         return None
-    return init_method.type
+
+    # case 1: normal FuncDef
+    if isinstance(init_method, FuncDef) and isinstance(init_method.type, CallableType):
+        init_node = typ.type.get("__init__") or typ.type.get(ATTRS_INIT_NAME)
+        if init_node is None or not init_node.plugin_generated:
+            return None
```

Comment:
OK, this now rejects any custom `__init__` methods straight away - probably too harsh. 

Since this seems to take longer (and more efforts) than just dropping another check on top of the existing logic, let's revert the `plugin_generated` check addition for now and work on that in a separate PR?